tools/pyi/gen_pyi.py

```diff
@@ -580,6 +584,31 @@ def gen_nn_functional(fm: FileManager) -> None:
     )
 
 
+def gather_docstrs() -> Dict[str, str]:
```

Comment:
nit: could you add a top level comment here explaining the strategy: 
- mock torch and torch._C which are the only dependencies of the docs files
- mock the _add_docstr function to save the docstring
- import _torch_docs and _tensor_docs to collect all the docstrings.